src/main/java/com/uid2/admin/vertx/service/SiteService.java

```diff
@@ -72,15 +72,36 @@ public void setupRoutes(Router router) {
             }
         }, Role.CLIENTKEY_ISSUER));
         router.post("/api/site/set-types").blockingHandler(auth.handle((ctx) -> {
-                    synchronized (writeLock) {
-                        this.handleSiteTypesSet(ctx);
-                    }
-                    }, Role.CLIENTKEY_ISSUER));
+            synchronized (writeLock) {
+                this.handleSiteTypesSet(ctx);
+            }
+        }, Role.CLIENTKEY_ISSUER));
         router.post("/api/site/domain_names").blockingHandler(auth.handle((ctx) -> {
             synchronized (writeLock) {
                 this.handleSiteDomains(ctx);
             }
         }, Role.CLIENTKEY_ISSUER));
+        handleSiteBackfillCreated();
+    }
+
+    private void handleSiteBackfillCreated() {
+        try {
+            final Collection<Site> sites = new ArrayList<>(this.siteProvider.getAllSites());
```

Comment:
Just curious why this needs to be mapped to an ArrayList ? `this.siteProvider.getAllSites()` returns a `Collection<Site>` which is iterable in a for loop.

Reply:
fixed